### DISCUSSION TITLE
Accessing Preference Array from Frontend Application Contribution

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#10144] [General] [answered]
Hi, 
I tried accessing "editor.rulers" using console.log from a Frontend Application Contribution. However, it only returns only the default value of the preference. However, if I try accessing it at a later stage say console logging after clicking on a menu item, it returns the correct values of the preference. I need help regarding the correct way to access the Array so that all the data in the array can be returned. 

Thanks

**Comment by msujew** [comment, marked as answer]
Hi @maheshrajrp,

The preference service takes a bit of time to be correctly initialized (i.e. loading values from user and workspace preferences). You can use the `ready` promise to execute any code that needs to be executed as soon as the preference service is ready.

```ts
@inject(PreferenceService)
protected readonly preferenceService: PreferenceService;
...
this.preferenceService.ready.then(() => {
    const preference = this.preferenceService.get(...);
    // Do some work with your preferences :)
});
```

**Reply by maheshrajrp** [reply]
Hi @msujew ,

Thanks. This solved it.
